zap/src/parser/reports.rs

```diff
@@ -338,6 +353,17 @@ impl Report<'_> {
 			Self::AnalyzeNamedReturn { name_span } => {
 				vec![Label::primary((), name_span.clone()).with_message("must be removed")]
 			}
+
+			Self::AnalyzeOrDuplicateType { prev_span, dup_span } => {
+				vec![
+					Label::secondary((), prev_span.clone()).with_message("previous type usage"),
+					Label::primary((), dup_span.clone()).with_message("duplicate type usage"),
+				]
+			}
+
+			Self::AnalyzeOrNonPrimitiveType { span } => {
+				vec![Label::primary((), span.clone()).with_message("non-primitive type")]
```

Comment:
This is unclear to me. I think we might just need to explain it better in the docs maybe.

Reply:
I don't think there's much I can do here to explain it better. A help note might improve it, but explaining it in the docs will probably be a better choice.